lib/watir/radio_set.rb

```diff
@@ -0,0 +1,231 @@
+    #
+    # @return [Boolean]
+    #
+
+    def disabled?
+      !enabled?
+    end
+
+    #
+    # Returns true if all radio buttons in the set are disabled.
+    #
+    # @return [String]
+    #
+
+    def name
+      @name ||= source.name
+    end
+
+    #
+    # If RadioSet exists, this always returns 'radio'.
+    #
+    # @return [String]
+    #
+
+    def type
+      source.send :assert_exists
+      'radio'
+    end
+
+    #
+    # Returns true if the radio set has one or more radio buttons where label matches the given value.
+    #
+    # @param [String, Regexp] str_or_rx
+    # @return [Boolean]
+    #
+
+    def include?(str_or_rx)
+      radio(label: str_or_rx).exist?
+    end
+
+    #
+    # Select the radio button whose value or label matches the given string.
+    #
+    # @param [String, Regexp] str_or_rx
+    # @raise [Watir::Exception::UnknownObjectException] if the Radio does not exist.
+    # @return [String] The value or text of the radio selected.
+    #
+
+    def select(str_or_rx)
+      found_by_value = radio(value: str_or_rx)
+      found_by_text = radio(label: str_or_rx)
+
+      if found_by_value.exist?
+        found_by_value.click unless found_by_value.selected?
+        return found_by_value.value
+      elsif found_by_text.exist?
+        found_by_text.click unless found_by_text.selected?
+        return found_by_text.text
+      else
+        raise UnknownObjectException, "Unable to locate radio matching #{str_or_rx.inspect}"
+      end
+    end
+
+    #
+    # Returns true if any of the radio button label matches the given value.
+    #
+    # @param [String, Regexp] str_or_rx
+    # @raise [Watir::Exception::UnknownObjectException] if the options do not exist
+    # @return [Boolean]
+    #
+
+    def selected?(str_or_rx)
+      found = frame.radio(label: str_or_rx)
+
+      return found.selected? if found.exist?
+      raise UnknownObjectException, "Unable to locate radio matching #{str_or_rx.inspect}"
+    end
+
+    #
+    # Returns the value of the selected radio button in the set.
+    # Returns nil if no radio is selected.
+    #
+    # @return [String, nil]
+    #
+
+    def value
+      sel = selected
+      sel && sel.value
+    end
+
+    #
+    # Returns the text of the selected radio button in the set.
+    # Returns nil if no option is selected.
+    #
+    # @return [String, nil]
+    #
+
+    def text
+      sel = selected
+      sel && sel.text
+    end
+
+    #
+    # Returns the selected Radio element.
+    # Returns nil if no radio button is selected.
+    #
+    # @return [Watir::Radio, nil]
+    #
+
+    def selected
+      find(&:selected?)
+    end
+
+    #
+    # Returns true if two elements are equal.
+    #
+    # @example
+    #   browser.radio_set(id: 'new_user_newsletter_yes') == browser.radio_set(id: 'new_user_newsletter_no')
+    #   #=> true
+    #
+
+    def ==(other)
+      return false unless other.kind_of?(self.class)
+      frame == other.frame
```

Comment:
This is checking that both `RadioSet` belong to the same `form` element? This would be incorrect when a form has 2 sets of radio buttons. Given the page:

~~~~~~~~html
<html>
  <body>
    <form>
      <input type="radio" name="a" id="a1"><label for="a1">my a_1</label>  
      <input type="radio" name="a" id="a2"><label for="a2">my a_2</label>
      <input type="radio" name="a" id="a3"><label for="a3">my a_3</label>

      <input type="radio" name="b" id="b1" value="b1v"><label for="b1">my b_1</label>  
      <input type="radio" name="b" id="b2" value="b2v"><label for="b2">my b_2</label>
      <input type="radio" name="b" id="b3" value="b3v"><label for="b3">my b_3</label>
    </form>
  </body>
</html>
~~~~~~~~

Comparing the 2 groups gives `true`:

~~~~~~~~ruby
b.radio_set(name: 'a') == b.radio_set(name: 'b')
#=> true    #--> should be false
~~~~~~~~

I think you want to compare the `source` instead - ie the line should be:'

~~~~~~~~ruby
source == other.source
~~~~~~~~